cratetorrent/src/metainfo.rs

```diff
@@ -187,18 +192,18 @@ impl Metainfo {
                     let url = Url::parse(&tracker)?;
                     // the tracker may be over UDP, which we don't support (yet)
                     if url.scheme() == "http" || url.scheme() == "https" {
-                        trackers.push((url, false));
+                        trackers.push(TrackerUrl { url, is_udp: false });
                     } else if url.scheme() == "udp" {
-                        trackers.push((url, true));
+                        trackers.push(TrackerUrl { url, is_udp: true });
```

Comment:
Something like this would be more Rust ergonomic:
```rust
match url.scheme() {
    "http" | "https" => trackers.push(TrackerUrl { url, is_udp: false }),
    "udp" => trackers.push(TrackerUrl { url, is_udp: true }),
    _ => {}
}
```

Reply:
I think it should be discussed out of this PR's scope, since it is refactoring of old code, not new one.

Reply:
Nothing to discuss here, this is just a suggestion to reduce the complexity, since the extra if/else statement was introduced by this PR.

Reply:
In earlier versions of rust pattern matching with anything except enums was slower than if/else. Not sure if it is fixed now

Reply:
http://wiki.c2.com/?PrematureOptimization

Reply:
I personally agree that it's easier to read, and that it's within the PR's scope (since I added said code).